tests/BlastDBNodeTest.php

```diff
@@ -0,0 +1,134 @@
+<?php
+namespace Tests;
+
+use StatonLab\TripalTestSuite\DBTransaction;
+use StatonLab\TripalTestSuite\TripalTestCase;
+
+/**
+ * Tests BlastDB Node CRUD
+ */
+class BlastDBNodeTest extends TripalTestCase {
+  // Uncomment to auto start and rollback db transactions per test method.
+  use DBTransaction;
+
+  /**
+   * BlastDB Node Type exists.
+   *
+   * Check that the BlastDB node type exists. It should be created
+   * when the module is installed by the Drupal Node API.
+   */
+  public function testBlastDBNodeExists() {
+
+    // Get a list of all types available.
+    $types = node_type_get_types();
+
+    // The BlastDB node type must be in the list.
+    $this->assertArrayHasKey('blastdb', $types);
+
+    // Additionally, the blastdb node type should be created by blast_ui.\
+    // This checks the case where the node type might be created by
+    // a different module.
+    $this->assertEquals($types['blastdb']->module, 'blast_ui');
+  }
+
+  /**
+   * Test Creating a BlastDB Node.
+   *
+   * Note: We can't test this by submitting the form via PUT because it requires
+   *  permission to access /node/add/blastdb; however, we don't yet have a
+   *  way to do this with TripalTestSuite. Furthermore, testing HTTP Requests
+   *  would not give us access to the data added via the test due to database
+   *  transactions.
+   */
+  public function testBlastDBNodeCreate() {
+    module_load_include('inc', 'node', 'node.pages');
+
+    // Log in the god user.
+    global $user;
+    $user = user_load(1);
+
+    $node = array('type' => 'blastdb');
+
+    // Fill in the form.
+    $form_state = array(
+      'values' => array(
+        'db_name' => 'Test Blast Database',
```

Comment:
So if i run this test on my site with a db that is Test Blast Database, what happens?  If it lets me create it, your assertion that num results = 1 is false.  If it wont let me create it, you'll get an error here.

What I do is use really stupid names in my tests (Banana Ostrich Blast Test) to reduce chance of these conflicts.  Test Blast Database is actually the sort of test i might make when trying to set up the module the first time...

Reply:
Good point!

Reply:
You can also use Faker to generate random strings :)

You can do that by adding this right under your `namespace`
```php
use Faker\Factory;
```

And then using it as such:
```php
$faker = Factory::create();

$name = $faker->name;
$someString = $faker->string;
````
If you'd like to know all available faker methods see: https://github.com/fzaninotto/Faker

Reply:
Oh, I will definitely use this for the create!

I would love to do this for the update too but since the blastdb is created in a dataseeder how would I know what to select in the test?

Reply:
Since test functions should be independent from each other (unless explicitly specified using PHPUnit doc blocks), you should actually create the and then update in the same function to test the update functionality. It's a bit repetitive but apparently that's the practice. 

If you run the create in the seeder, you can always add a method to the seeder to return the created object (which you have saved into a private property) since it's just a PHP class :)